### PR TITLE
Suggested Changes

### PR DESCRIPTION
**Pull request by f3rn0s**
Just some stuff to clean up the code :).

Love your other projects btw. :P

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -6,12 +6,15 @@ use cmp::Ordering;
 
 fn main() {
     println!("Guess the number!\n");
+
     let secret = thread_rng().gen_range(1, 101);
 
-    let mut is_match = false;
-    while !is_match {
+    loop {
```
Comment (f3rn0s) [inline src/main.rs:12]: Uses loop which is more rusty.

File: src/main.rs [hunk 2]
```diff
@@ -21,18 +24,23 @@ fn match_input (guess: u32, secret: u32) -> bool {
         Ordering::Greater => println!("lower!"),
         Ordering::Equal   => println!("bingo! 🎉"),
     }
-    return guess.eq(&secret);
```
Comment (f3rn0s) [inline src/main.rs:24]: Return can, and should, be mostly avoided in rust.

File: src/main.rs [hunk 3]
```diff
@@ -21,18 +24,23 @@ fn match_input (guess: u32, secret: u32) -> bool {
...
+
+    guess.eq(&secret)
 }
 
 fn ask_number () -> u32 {
-    println!("Please input your guess : ");
+    loop {
```
Comment (f3rn0s) [inline src/main.rs:32]: Created a loop, where in the question will be repeated until the user provides a guess.

File: src/main.rs [hunk 4]
```diff
@@ -21,18 +24,23 @@ fn match_input (guess: u32, secret: u32) -> bool {
...
-    let mut guess = String::new();
-    io::stdin().read_line(&mut guess).unwrap();
+        let mut guess = String::new();
+        io::stdin().read_line(&mut guess).unwrap();
 
-    // parse to a number
-    let guess: u32 = guess.trim().parse()
-        .expect("You need to input a number :(");
+        // parse to a number
+        let guess: u32 = match guess.trim().parse() {
```
Comment (f3rn0s) [inline src/main.rs:39]: Using a match statement is a great way to implement error handling.

File: src/main.rs [hunk 5]
```diff
@@ -21,18 +24,23 @@ fn match_input (guess: u32, secret: u32) -> bool {
...
+            Ok(number) => number,
+            Err(_) => continue
```
Comment (f3rn0s) [inline src/main.rs:41]: If an error occurs now, we ignore it and retry the loop.

**Comment by gillesdemey** [comment]
Thanks for the helpful feedback! 😎

**Pull request merged**